src/wallet_sync.rs

```diff
@@ -91,127 +93,212 @@ const WATCH_ONLY_SWAPCOIN_LABEL: &str = "watchonly_swapcoin_label";
 //swapcoins are UTXOs + metadata which are not from the deterministic wallet
 //they are made in the process of a coinswap
 #[derive(serde::Serialize, serde::Deserialize, Debug, Clone)]
-pub struct WalletSwapCoin {
+pub struct IncomingSwapCoin {
     pub my_privkey: SecretKey,
     pub other_pubkey: PublicKey,
     pub other_privkey: Option<SecretKey>,
     pub contract_tx: Transaction,
     pub contract_redeemscript: Script,
-    //either timelock_privkey for outgoing swapcoins or hashlock_privkey for incoming swapcoins
-    pub contract_privkey: SecretKey,
+    pub hashlock_privkey: SecretKey,
     pub funding_amount: u64,
     pub others_contract_sig: Option<Signature>,
     pub hash_preimage: Option<[u8; 32]>,
 }
-//TODO split WalletSwapCoin into two structs, IncomingSwapCoin and OutgoingSwapCoin
-//where Incoming has hashlock_privkey and Outgoing has timelock_privkey
-//that is a much more rustic way of doing things, which uses the compiler to check for some bugs
 
-impl WalletSwapCoin {
+//swapcoins are UTXOs + metadata which are not from the deterministic wallet
+//they are made in the process of a coinswap
+#[derive(serde::Serialize, serde::Deserialize, Debug, Clone)]
+pub struct OutgoingSwapCoin {
+    pub my_privkey: SecretKey,
+    pub other_pubkey: PublicKey,
+    pub other_privkey: Option<SecretKey>,
```

Comment:
OutgoingSwapCoin will *never* have the `other_privkey`. That other privkey can only ever be given to us via private key handover (see `handle_private_key_handover`), which only ever happens for IncomingSwapCoins.

Reply:
Removed